overlord/restart/restart.go

```diff
@@ -157,6 +254,67 @@ func Request(st *state.State, t RestartType, rebootInfo *boot.RebootInfo) {
 	rm.handleRestart(t, rebootInfo)
 }
 
+func changeHeldForSystemRestart(chg *state.Change) (int, error) {
+	var count int
+	err := chg.Get("held-for-system-restart", &count)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return 0, err
+	}
+	return count, nil
+}
+
+func incrementHeldForSystemRestart(chg *state.Change) error {
+	if chg == nil {
+		// nothing to do
+		return nil
+	}
+	count, err := changeHeldForSystemRestart(chg)
+	if err != nil {
+		return err
+	}
+	chg.Set("held-for-system-restart", count+1)
+	return nil
+}
+
+// FinishTaskWithRestart will finish a task that needs a restart, by
+// setting its status and requesting a restart.
+// It should usually be invoked returning its result immediately
+// from the caller.
+// In some situations (classic) it might return state.Hold instead
+// to held progress until a non-automatic system restart occurs.
```

Comment:
Alberto's suggestion looks good just wanted to point out a minor typo in case a mix of the two is committed.
```suggestion
// to hold progress until a non-automatic system restart occurs.
```